### PR TITLE
Products model added and configured as an ActiveAdmin resource

### PR DESCRIPTION
**Pull request by Stefano-loop**
#### :link: Board reference:

* [As an admin user I should be able to create products](https://loopstudio.atlassian.net/jira/software/projects/STEF/boards/42?selectedIssue=STEF-6)
* [As an admin user I should be able to update products](https://loopstudio.atlassian.net/jira/software/projects/STEF/boards/42?selectedIssue=STEF-7)
* [As an admin user I should be able to delete products](https://loopstudio.atlassian.net/jira/software/projects/STEF/boards/42?selectedIssue=STEF-8)
* [As an admin user I should be able to get a list of products](https://loopstudio.atlassian.net/jira/software/projects/STEF/boards/42?selectedIssue=STEF-1)

---

#### Description:

Products added and configured as an **ActiveAdmin** resource. Admin users can list, create, update and delete products.

**Products:**

- Have a Name, Description, Price and Stock. All fields are mandatory. 
- **Name** and **description** are both Strings with no validation except for presence.
- **Price** accepts decimal values were **stock** its just an integer. Validation checks for negative values in both cases.

---

#### :heavy_check_mark:Tasks:

- Products model added with name, description, price and stock as attributes.

- Migration created. It adds products table to the database.

- Products registered as an ActiveAdmin resource by adding 'products.rb' file to /app/admin folder.

- Name, description, price and stock added as permitted parameters in perviously mentioned file.

- Specs updated to contemplate products model with its corresponding validations.

- AdminUser model test fixed. Missing describe block was added. (spec/models/admin_user_spec.rb)

---

### CODE REVIEW

File: db/migrate/20200117134857_create_products.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateProducts < ActiveRecord::Migration[6.0]
+  def change
+    create_table :products do |t|
+      t.string :name
+      t.string :description
+      t.decimal :price
+      t.integer :stock
```
Comment (letiesperon) [inline db/migrate/20200117134857_create_products.rb]: not null everywhere please, except description maybe

File: db/migrate/20200117134857_create_products.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+      t.timestamps
+    end
+  end
```
Comment (letiesperon) [inline db/migrate/20200117134857_create_products.rb]: Think about whether you should be adding any index and explain your decision

File: spec/factories/products.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+
+FactoryBot.define do
+  factory :product do
+    name { Faker::Commerce.product_name }
+    description { Faker::Lorem.sentence }
+    price { Faker::Commerce.price }
+    stock { Faker::Number.number }
```
Comment (letiesperon) [inline spec/factories/products.rb]: this can give you negative

File: db/migrate/20200117134857_create_products.rb (hunk 2 above)
Comment (Stefano-loop) [reply db/migrate/20200117134857_create_products.rb]: Users are going to search by name/description most of the time, so maybe adding an index with those columns would improve performance.

Comment (letiesperon) [reply db/migrate/20200117134857_create_products.rb]: not on description but on name, yes

**Pull request merged**